hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteWorkerActor.scala

```diff
@@ -1,47 +1,94 @@
 package cromwell.services.metadata.hybridcarbonite
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import cats.data.NonEmptyList
+import cromwell.core.retry.SimpleExponentialBackoff
+import cromwell.core.{WorkflowAborted, WorkflowFailed, WorkflowId, WorkflowSucceeded}
+import cromwell.services.metadata.MetadataArchiveStatus.Unarchived
+import cromwell.services.metadata.MetadataService.{QueryForWorkflowsMatchingParameters, WorkflowQueryFailure, WorkflowQuerySuccess}
+import cromwell.services.metadata.WorkflowQueryKey._
+import cromwell.services.metadata.hybridcarbonite.CarboniteWorkerActor.CarboniteWorkflowComplete
+import cromwell.services.metadata.hybridcarbonite.CarbonitingMetadataFreezerActor.FreezeMetadata
 import cromwell.util.GracefulShutdownHelper
 import cromwell.util.GracefulShutdownHelper.ShutdownCommand
-import cromwell.services.metadata.hybridcarbonite.CarboniteWorkerActor.DoCarboniting
 
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 
-class CarboniteWorkerActor(carboniteInterval: FiniteDuration, ioActor: ActorRef) extends Actor with ActorLogging with GracefulShutdownHelper {
+class CarboniteWorkerActor(carboniterConfig: HybridCarboniteConfig,
+                           serviceRegistryActor: ActorRef,
+                           ioActor: ActorRef) extends Actor with ActorLogging with GracefulShutdownHelper {
 
   implicit val ec: ExecutionContext = context.dispatcher
 
-  scheduleNextCarboniting()
+  val carboniteFreezerActor = context.actorOf(CarbonitingMetadataFreezerActor.props(carboniterConfig, self, serviceRegistryActor, ioActor))
+
+  val findWorkflowToCarboniteQuery: Seq[(String, String)] = Seq(
+    IncludeSubworkflows.name -> "false",
+    Status.name -> WorkflowSucceeded.toString,
+    Status.name -> WorkflowFailed.toString,
+    Status.name -> WorkflowAborted.toString,
+    MetadataArchiveStatus.name -> Unarchived.toString,
+    Page.name -> "1",
+    PageSize.name -> "1"
+  )
+
+  val backOff = SimpleExponentialBackoff(
+    initialInterval = 5.seconds,
+    maxInterval = 5.minutes,
+    multiplier = 1.1,
+    randomizationFactor = 0.0
+  )
+
+  scheduleNextCarboniting(backOff.backoffMillis)
+
 
   override def receive: Receive = {
-    case DoCarboniting => doCarboniting()
-    case ShutdownCommand => stopGracefully()
-    case oops => log.error(s"Programmer Error! The CarboniteWorkerActor is not to be talked to! ($sender sent $oops})")
+    case s: WorkflowQuerySuccess => {
+      if (s.response.results.nonEmpty) carboniteWorkflow(s.response.results.head.id)
+      else scheduleNextCarboniting(backOff.backoffMillis)
+    }
+    case f: WorkflowQueryFailure => {
+      log.error(s"Something went wrong while querying to find workflow to carbonite. It will retry in sometime. Error: ${f.reason}")
```

Comment:
TOL and also a question for the broader audience: is "carbonite" the terminology we want to surface?

```suggestion
      log.error(f, s"Error querying workflows to carbonite, will retry")
```